### ISSUE TITLE
ShardInfo in NodeInfo MXBean is missing the schema name leading to duplicates

### ISSUE THREAD

**Issue by proddata** [crate/crate#16450]
### CrateDB version

any after #10835

### CrateDB setup information

CrateDB + JMX Exporter (or another way to read the JMX metrics)

### Problem description

[ShardInfo in NodeInfo MXBean](https://cratedb.com/docs/crate/reference/en/latest/admin/monitoring.html#nodeinfo-mxbean) is missing the schema name leading to duplicates when exporting why jmx exporter.

This seems to be an oversight of https://github.com/crate/crate/pull/10835
Adding a schema will probably also need an update of the jmx exporter.

### Steps to Reproduce

```sql
CREATE TABLE s01.tab (i INT) CLUSTERED INTO 1 SHARDS;
CREATE TABLE s02.tab (i INT) CLUSTERED INTO 1 SHARDS;
INSERT INTO s01.tab (i) VALUES (1);
INSERT INTO s02.tab (i) VALUES (1);
```

+ jmx exporter + Prometheus setup

### Actual Result

`name="shard_info",property="size",id="0",table="tab",partition_ident="",`
`name="shard_info",property="size",id="0",table="tab",partition_ident="",`

### Expected Result

`name="shard_info",property="size",id="0",table="tab",schema="s01",partition_ident="",`
`name="shard_info",property="size",id="0",table="tab",schema="s02",partition_ident="",`

**Comment by matriv** [comment]
Thx for finding out the root cause of those errors in prometheus!

**Comment by matriv** [comment]
@proddata Thx for finding out the root cause of those errors in prometheus!

Keep in mind that jmx_exported can probably handle the new attribute (schema) without any change, but please test it to verify that this is the case.

**Comment by mfussenegger** [comment]
@proddata Could you test with dev builds containing both https://github.com/crate/jmx_exporter/pull/73 and https://github.com/crate/crate/pull/16456?

**Comment by proddata** [comment]
> @proddata Could you test with dev builds containing both [crate/jmx_exporter#73](https://github.com/crate/jmx_exporter/pull/73) and #16456?

Tested with CrateDB on **master** and **4.7.2** with jmx_exporter on **master** and **1.1.0**.
Works as expected. ✅ 

---

I assume there is no meaningful mitigation for jmx_exporter to handle duplicates for older versions of CrateDB? i.e. adding a ordinal.

**Comment by mfussenegger** [comment]
> I assume there is no meaningful mitigation for jmx_exporter to handle duplicates for older versions of CrateDB? i.e. adding a ordinal.

Unfortunately no. 

---

Released crate-jmx-exporter 1.2.0
